flaky QLOGDIR test on Windows

```
 === RUN   TestQlogDirEnvironmentVariable
    qlog_dir_test.go:23: 
        	Error Trace:	D:/a/quic-go/quic-go/integrationtests/self/qlog_dir_test.go:23
        	            				C:/hostedtoolcache/windows/go/1.23.4/x64/src/testing/testing.go:1176
        	            				C:/hostedtoolcache/windows/go/1.23.4/x64/src/testing/testing.go:1354
        	            				C:/hostedtoolcache/windows/go/1.23.4/x64/src/testing/testing.go:1684
        	Error:      	Received unexpected error:
        	            	remove C:\Users\RUNNER~1\AppData\Local\Temp\temp_test_dir458385285\qlogs\69bb49ec7519e0f819eabf9125f24b3e9def3207_server.sqlog: The process cannot access the file because it is being used by another process.
        	Test:       	TestQlogDirEnvironmentVariable
--- FAIL: TestQlogDirEnvironmentVariable (0.01s)

--- a/integrationtests/self/qlog_dir_test.go
+++ b/integrationtests/self/qlog_dir_test.go
@@ -15,15 +15,9 @@ import (
 )
 
 func TestQlogDirEnvironmentVariable(t *testing.T) {
-	originalQlogDirValue := os.Getenv("QLOGDIR")
-	tempTestDirPath, err := os.MkdirTemp("", "temp_test_dir")
-	require.NoError(t, err)
-	t.Cleanup(func() {
-		require.NoError(t, os.Setenv("QLOGDIR", originalQlogDirValue))
-		require.NoError(t, os.RemoveAll(tempTestDirPath))
-	})
-	qlogDir := path.Join(tempTestDirPath, "qlogs")
-	require.NoError(t, os.Setenv("QLOGDIR", qlogDir))
+	tempDir := t.TempDir()
+	qlogDir := path.Join(tempDir, "qlogs")
+	t.Setenv("QLOGDIR", qlogDir)
 
 	serverStopped := make(chan struct{})
 	server, err := quic.Listen(
@@ -60,7 +54,7 @@ func TestQlogDirEnvironmentVariable(t *testing.T) {
 	server.Close()
 	<-serverStopped
 
-	_, err = os.Stat(tempTestDirPath)
+	_, err = os.Stat(qlogDir)
 	qlogDirCreated := !os.IsNotExist(err)
 	require.True(t, qlogDirCreated)
 

